### PR TITLE
nixpkgs-lint-community: init at 0.2.1

### PR DESCRIPTION
**Pull request by figsoda**
https://github.com/nix-community/nixpkgs-lint

not sure if this is the best name to use, but I ran out of ideas

cc @Artturin @siraben 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Why is there a top-level entry for https://github.com/NixOS/nixpkgs/blob/master/maintainers/scripts/nixpkgs-lint.nix ? I don't think we have that for most other tools and I would just vote to drop that.

File: pkgs/tools/nix/nixpkgs-lint/Cargo.lock [hunk 1]
```diff
@@ -0,0 +1,715 @@
...
+checksum = "d4423c784fe11398ca91e505cdc71356b07b1a924fc8735cfab5333afe3e18bc"
+dependencies = [
+ "cc",
+ "regex",
+]
+
+[[package]]
+name = "tree-sitter-nix"
+version = "0.0.1"
+source = "git+https://github.com/cstrahan/tree-sitter-nix#1b69cf1fa92366eefbe6863c184e5d2ece5f187d"
```
Comment (SuperSandro2000) [inline pkgs/tools/nix/nixpkgs-lint/Cargo.lock:579]: This got moved to nix-community. Can't we create a tag there to avoid vendoring the cargo.lock file?

**Comment by figsoda** [comment]
Looks like this is not the only one, there is also `nix-generate-from-cpan` which is documented in the manual

I'm not sure if anyone still uses that nixpkgs-lint though

File: pkgs/tools/nix/nixpkgs-lint/Cargo.lock (hunk 1 above)
Comment (figsoda) [reply pkgs/tools/nix/nixpkgs-lint/Cargo.lock]: nixpkgs-lint got a new release and I've removed the Cargo.lock now

**Comment by SuperSandro2000** [comment]
> I'm not sure if anyone still uses that nixpkgs-lint though

It seems to be rather dead to me, I would just squat the name,

**Comment by figsoda** [comment]
#236766 we'll see how it goes

**Comment by figsoda** [comment]
Should we go with the current approach instead?

File: pkgs/tools/nix/nixpkgs-lint/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  };
+
+  cargoHash = "sha256-LWtBO0Ai5cOtnfZElBrHZ7sDdp3ddfcCRdTA/EEDPfE=";
+
+  meta = with lib; {
+    description = "A fast semantic linter for Nix using tree-sitter";
+    homepage = "https://github.com/nix-community/nixpkgs-lint";
+    changelog = "https://github.com/nix-community/nixpkgs-lint/releases/tag/${src.rev}";
+    license = licenses.mit;
+    maintainers = with maintainers; [ figsoda ];
```
Comment (Artturin) [inline pkgs/tools/nix/nixpkgs-lint/default.nix:24]:
```suggestion
    maintainers = with maintainers; [ figsoda artturin ];
```

**Comment by Artturin** [comment]
Let's go with this for now

**Pull request merged**